rules/apps.libsonnet

```diff
@@ -63,20 +63,26 @@
           {
             record: 'namespace:kube_pod_container_resource_requests_memory_bytes:sum',
             expr: |||
-              sum by (namespace, label_name) (
-                  sum(kube_pod_container_resource_requests_memory_bytes{%(kubeStateMetricsSelector)s} * on (endpoint, instance, job, namespace, pod, service) group_left(phase) (kube_pod_status_phase{phase=~"Pending|Running"} == 1)) by (namespace, pod)
-                * on (namespace, pod)
-                  group_left(label_name) kube_pod_labels{%(kubeStateMetricsSelector)s}
+              sum by (namespace) (
+                  sum by (namespace, pod) (
+                      kube_pod_container_resource_requests_memory_bytes{%(kubeStateMetricsSelector)s}
```

Comment:
The many-to-many errors we see typically happen when there is a rollout of kube-state-metrics, in a situation where there is the possibility for momentarily scraping 2 instances of kube-state-metrics exposing the same set of metrics twice, which would result in this sum being incorrect as it is. So in addition to have we already have for the pod phase metric, we also need to `max by(namespace, pod, container) (kube_pod_container_resource_requests_memory_bytes{%(kubeStateMetricsSelector)s})` here.